- Description: iucv: potential memory leek
  Symptom:     available memory decreases over time
  Problem:     missing kfree(new_handler) in iucv_register_program()
  Solution:    added kfree(new_handler)
  Archs:       s390-31, s390-64
  Problem-ID:  --

--- linux-2.5/drivers/s390/net/iucv.c	22 Mar 2004 07:43:43 -0000	1.27
+++ linux-2.5/drivers/s390/net/iucv.c	15 Apr 2004 09:18:47 -0000	1.27.2.1
@@ -1,5 +1,5 @@
 /* 
- * $Id: iucv.c,v 1.27 2004/03/22 07:43:43 braunu Exp $
+ * $Id: iucv.c,v 1.27.2.1 2004/04/15 09:18:47 braunu Exp $
  *
  * IUCV network driver
  *
@@ -29,7 +29,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  *
- * RELEASE-TAG: IUCV lowlevel driver $Revision: 1.27 $
+ * RELEASE-TAG: IUCV lowlevel driver $Revision: 1.27.2.1 $
  *
  */
 
@@ -351,7 +351,7 @@
 static void
 iucv_banner(void)
 {
-	char vbuf[] = "$Revision: 1.27 $";
+	char vbuf[] = "$Revision: 1.27.2.1 $";
 	char *version = vbuf;
 
 	if ((version = strchr(version, ':'))) {
@@ -800,6 +800,7 @@
 		if (iucv_pathid_table == NULL) {
 			printk(KERN_WARNING "%s: iucv_pathid_table storage "
 			       "allocation failed\n", __FUNCTION__);
+			kfree(new_handler);
 			return NULL;
 		}
 		memset (iucv_pathid_table, 0, max_connections * sizeof(handler *));

